WooCommerce/Classes/ViewRelated/Orders/Order Filters/Order Status Filter/OrderStatusFilterViewController.swift

```diff
@@ -0,0 +1,206 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureMainView()
+        configureRows()
+        configureTableView()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension OrderStatusFilterViewController {
+
+    func configureNavigationBar() {
+        title = Localization.navigationBarTitle
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        /// Registers all of the available TableViewCells
+        ///
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+    }
+
+    func configureRows() {
+        rows = [.any, .pending, .processing, .onHold, .failed, .cancelled, .completed, .refunded]
+    }
+
+    func selectOrDelesectRow(_ row: Row) {
+        guard let status = row.status else {
+            return
+        }
+        if selected.contains(status) {
+            selected.removeAll { $0 == status }
+        }
+        else {
+            selected.append(status)
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension OrderStatusFilterViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension OrderStatusFilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row] {
+        case .any:
+            selected = []
+            onCompletion(selected)
+            tableView.reloadData()
+        default:
+            selectOrDelesectRow(rows[indexPath.row])
+            onCompletion(selected)
+            tableView.reloadData()
+            return
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+}
+
+// MARK: - Cell configuration
+//
+private extension OrderStatusFilterViewController {
+    enum Row: CaseIterable {
+        case any
+        case pending
+        case processing
+        case onHold
+        case failed
+        case cancelled
+        case completed
+        case refunded
```

Comment:
Nit: how about keeping this simpler by just having 2 cases: `any` and `status(value: OrderStatusEnum)`? Or `Row` can simply be a struct that accepts an optional `OrderStatusEnum` as input. WDYT?

Reply:
I prefer to not use `OrderStatusEnum` here because of the `custom` type in `OrderStatusEnum` and also because I believe there are no big benefits in doing that, or I'm wrong?